src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py

```diff
@@ -141,11 +109,40 @@ def uploadWorker(workInput, results, dbsUrl):
     return
 
 
+def isPassiveError(exceptionObj):
+    """
+    This function will parse the exception object and decide whether
+    the error should be dealt with in a passive way, or if it's a
+    hard error and the component better crash
+    :param exceptionObj:
+    :return: a tuple with a boolean (passive or not) and a string with
+        the error message
+    """
+    passException = True
+    passiveErrorMsg = ['Service Unavailable', 'Service Temporarily Unavailable',
+                       'Proxy Error', 'Error reading from remote server',
+                       'Connection refused', 'timed out', 'Could not resolve',
+                       'OpenSSL SSL_connect: SSL_ERROR_SYSCALL']
+
+    excReason = getattr(exceptionObj, 'reason', '')
+    errorMsg = 'Failed to fetch parentage map from WMStats, skipping this cycle. '
+    errorMsg += 'Exception: {}. Reason: {}. Error: {}. '.format(type(exceptionObj).__name__,
+                                                                excReason, str(exceptionObj))
+    for passiveMsg in passiveErrorMsg:
+        if passiveMsg in excReason:
+            break
+        elif passiveMsg in str(exceptionObj):
```

Comment:
Matching a string inside the whole (string converted) exception object only to identify the type of Error, seems to me kind of error prone. I am not sure if this is a common practice, though. Mybe it is just me not being familiar with it.

Reply:
I don't think it's a good practice, but given that we do not have different exception types (or return code) for these exceptions, the only way we can prevent the component from crashing from transient issues in the system, is by parsing the error message returned in the HTTP call.

Reply:
Aren't all those expected to be visible inside the Exception Reason. And am I correct in the observation that in the current list of errors you consider, there are not only WMCore exceptions but also exceptions from libraries with regular use all over the place like SSL etc. Meaning we are not in full control of where the Error string would appear.

Reply:
Exception reason contains the string with the error message. IF, our request response comes back with a header.reason attribute:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Services/pycurl_manager.py#L285

however, I think there could be other problems when executing `curl.perform()` which would actually raise an exception right there, thus potentially not having the `reason` attribute.
Reading the traceback I posted in the GH issue, this is exactly what happened two days ago.